test/unit/extra_fields_test.rb

```diff
@@ -47,233 +42,213 @@ class ExtraFieldsTest < ActiveSupport::TestCase
-        assert_equal expected, @buyer.extra_fields
-      end
-
-      should 'not set key value not defined in FieldsDefinition' do
-        @buyer.extra_fields = { :non_existant => "is set" }
-        @buyer.save!
-        assert @buyer.extra_fields.empty?
-      end
-
-      #beware of this!!!
-      # should 'not set key value not defined in FieldsDefinition using [] notation' do
-      #   @buyer[:extra_fields][:hack] = "attack"
-      #   @buyer.save!
-      #   assert @buyer.extra_fields[:hack].nil?
-      # end
-
-      should 'not remove already set extra_fields' do
-        bar_field = FactoryBot.create(:fields_definition, :account => @provider,
-                            :target => "Account", :name => "deleted_field")
-        @buyer.reload
-        expected = @buyer.extra_fields = { :deleted_field => "exists yet" }
-        @buyer.save!
-        bar_field.destroy
-
-        @buyer.extra_fields = { :provider_extra_field => "bar" }
-        @buyer.save!
-        assert_equal 'exists yet', @buyer.extra_fields[:deleted_field]
-      end
-
-      should 'override using [] notation' do
-        @buyer.extra_fields = { :provider_extra_field => "[] notation overridable" }
-        @buyer.save!
-
-        @buyer[:extra_fields] = { }
-        @buyer.save!
-        assert @buyer.extra_fields.empty?
-      end
-
-      should 'force encoding on strings' do
-        @buyer.extra_fields = {
-          provider_extra_field: "\xD0\xBF\xD0\xBE\xD0\xBA\xD0\xB0\xD0\xB7\xD1\x8B\xD0\xB2\xD0\xB0\xD1\x82\xD1\x8C \xD0\xBD\xD0\xB0 \xD0\xB1\xD0\xB5\xD1\x81\xD0\xBF\xD0\xBB\xD0\xB0\xD1\x82\xD0\xBD\xD0\xBE\xD0\xBC \xD0\xBF\xD1\x80\xD0\xB8\xD0\xBB\xD0\xBE\xD0\xB6\xD0\xB5\xD0\xBD\xD0\xB8\xD0\xB8 \xD1\x82\xD0\xB0\xD0\xB1\xD0\xBB\xD0\xBE \xD0\xB2\xD1\x8B\xD0\xBB\xD0\xB5\xD1\x82\xD0\xBE\xD0\xB2 \xD1\x81\xD0\xB0\xD0\xBC\xD0\xBE\xD0\xBB\xD0\xB5\xD1\x82\xD0\xBE\xD0\xB2".force_encoding('BINARY')
-        }
-
-        str = @buyer.extra_fields[:provider_extra_field]
-
-        assert_equal Encoding.default_internal, str.encoding
-      end
-
-      should 'force encoding on the getter' do
-        @buyer.extra_fields[:provider_extra_field] = "\xD0\xBF\xD0\xBE\xD0\xBA\xD0\xB0\xD0\xB7\xD1\x8B\xD0\xB2\xD0\xB0\xD1\x82\xD1\x8C \xD0\xBD\xD0\xB0 \xD0\xB1\xD0\xB5\xD1\x81\xD0\xBF\xD0\xBB\xD0\xB0\xD1\x82\xD0\xBD\xD0\xBE\xD0\xBC \xD0\xBF\xD1\x80\xD0\xB8\xD0\xBB\xD0\xBE\xD0\xB6\xD0\xB5\xD0\xBD\xD0\xB8\xD0\xB8 \xD1\x82\xD0\xB0\xD0\xB1\xD0\xBB\xD0\xBE \xD0\xB2\xD1\x8B\xD0\xBB\xD0\xB5\xD1\x82\xD0\xBE\xD0\xB2 \xD1\x81\xD0\xB0\xD0\xBC\xD0\xBE\xD0\xBB\xD0\xB5\xD1\x82\xD0\xBE\xD0\xB2".force_encoding('BINARY')
-
-        str = @buyer.extra_fields[:provider_extra_field]
-
-        assert_equal Encoding.default_internal, str.encoding
-      end
-    end # setters
-
-  end # extra fields
-
-  context 'fields definitions' do
-    #TODO: do this test in user and cinstance also or find a way of doing
-    # shared_examples
+  class ExtrafieldsValidations < ActiveSupport::TestCase
+    setup do
```

Comment:
Perhaps more sets of tests could be wrapped in subclasses of `ExtraFieldsTest` and leave `#setup` to be defined only once at the parent class?

Reply:
yes, that is the goal @guicassolato https://github.com/3scale/porta/pull/1609#discussion_r374360551
Only not done yet but it is the plan :smile: